src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,935 @@
+                return callStatus;
+            } else if ("CONNECTED".equals(getJsonProperty(response, "state").asText())) {
+                callStatus.setCallId(getJsonProperty(response, "id").asText());
+                callStatus.setCallStatusState(CallStatus.CallStatusState.Connected);
+                return callStatus;
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+        return null;
+    }
+
+    @Override
+    public MuteStatus retrieveMuteStatus() throws Exception {
+        controlOperationsLock.lock();
+        try {
+            Boolean muted = doGet(buildHttpUrl(AUDIO_MUTED), Boolean.class);
+            if (muted) {
+                return MuteStatus.Muted;
+            } else {
+                return MuteStatus.Unmuted;
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public void sendMessage(PopupMessage popupMessage) throws Exception {
+    }
+
+    @Override
+    public void mute() throws Exception {
+        controlOperationsLock.lock();
+        try {
+            if (!doPost(AUDIO_MUTED, true).getStatusCode().is2xxSuccessful()) {
+                throw new CommunicationException("Unable to mute the device.");
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public void unmute() throws Exception {
+        controlOperationsLock.lock();
+        try {
+            if (!doPost(AUDIO_MUTED, false).getStatusCode().is2xxSuccessful()) {
+                throw new CommunicationException("Unable to unmute the device.");
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public List<Statistics> getMultipleStatistics() throws Exception {
+        ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+        EndpointStatistics endpointStatistics = new EndpointStatistics();
+
+        controlOperationsLock.lock();
+        try {
+            if (isValidControlCoolDown() && localStatistics != null && localEndpointStatistics != null) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Device is occupied. Skipping statistics refresh call.");
+                }
+                extendedStatistics.setStatistics(localStatistics.getStatistics());
+                extendedStatistics.setControllableProperties(localStatistics.getControllableProperties());
+
+                endpointStatistics.setInCall(localEndpointStatistics.isInCall());
+                endpointStatistics.setCallStats(localEndpointStatistics.getCallStats());
+                endpointStatistics.setVideoChannelStats(localEndpointStatistics.getVideoChannelStats());
+                endpointStatistics.setAudioChannelStats(localEndpointStatistics.getAudioChannelStats());
+
+                return Arrays.asList(extendedStatistics, endpointStatistics);
+            }

```

Comment:
It is made specifically for returning "cached" versions of statistics. The thing is that this adapter has a set of controllable properties, activating any control would result in emergency delivery situation, which will lead to the immediate REST Api calls every time the control is activated. If the control is activated like... 10 times in a row, it will result with stacking up calls and eventually - an error, because Symphony would not receive an expected response in time. That's why this caching mechanism is introduced - if the device is considered "occupied" - monitoring cycle is "skipped" and cached data is populated instead